cpu/sam0_common/periph/timer.c

```diff
@@ -235,6 +235,63 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
     return 0;
 }
 
+int timer_set(tim_t tim, int channel, unsigned timeout)
+{
+    DEBUG("Setting timer %i channel %i to now + %i\n", tim, channel, timeout);
+
+    unsigned max = UINT16_MAX;
+    const uint16_t flags = timer_config[tim].flags;
+
+    if (flags & TC_CTRLA_MODE_COUNT32) {
+        max = UINT32_MAX;
+    }
+
+    if (flags & TC_CTRLA_MODE_COUNT8) {
+        max = UINT8_MAX;
+    }
+
+    unsigned irq_state = irq_disable();
+    unsigned value = timeout + timer_read(tim);
+
+    /* set timeout value */
+    switch (channel) {
+    case 0:
+        dev(tim)->INTFLAG.reg = TC_INTFLAG_MC0;
+        _set_cc(tim, 0, value);
+        dev(tim)->INTENSET.reg = TC_INTENSET_MC0;
+        break;
+    case 1:
+        dev(tim)->INTFLAG.reg = TC_INTFLAG_MC1;
+        _set_cc(tim, 1, value);
+        dev(tim)->INTENSET.reg = TC_INTENSET_MC1;
+        break;
+    default:
+        return -1;
+    }
+
+    set_oneshot(tim, channel);
+
+    if (((value - timer_read(tim)) & max) > timeout) {
+        /* timer already fired, check if IRQ flag is set */
+        uint32_t mask = (channel == 0) ? TC_INTFLAG_MC0 : TC_INTFLAG_MC1;
+        if (!(mask & dev(tim)->INTFLAG.reg)) {
+            /* Hack: We stop the timer when setting the target to avoid a race.
+             * Sadly, the count is lost on stop, so we need to restore it
+             * afterwards. We add one to risk rather jumping over one tick
+             * rather than having the clock going backwards by one time */
+            unsigned now = timer_read(tim);
+            timer_stop(tim);
```

Comment:
TODO: Does this have nasty side affects like loosing the IRQ mask or the CC reg of the other channel?

Reply:
Is this TODO required to be done for this PR ?
If not, I can have a look at it.

Reply:
Ideally I would just extend the periph_timer test so that such side effects would be directly detected. I'll try to allocate some time for that, but currently my work load is quite high :-/

Reply:
I feel your pain, I'm somehow in the same position :)
So should we wait a bit for this PR ?